libs/runtime/detect.go

```diff
@@ -0,0 +1,14 @@
+package runtime
+
+import (
+	"context"
+
+	"github.com/databricks/cli/libs/env"
+)
+
+const envDatabricksRuntimeVersion = "DATABRICKS_RUNTIME_VERSION"
+
+func RunsOnDatabricks(ctx context.Context) bool {
```

Comment:
+1 for a common lib function. I'd expect this to also check for the existence of /Workspace though to avoid false positives. There may be all kinds of reasons why customers set the env var locally.

Reply:
I'd prefer not to test for `/Workspace` as it makes the code using it harder to test

Reply:
Yeah, testing is a pain :( We can leave it out, but then we should at least emit a debug log message whenever there's a match. There will be false positives.

Reply:
Testing is not an issue as long as you don't inline these expectations.

You can store a bool on the context that stores whether or not we're running on DBR (see `bundle/context.go` for inspiration). In tests, you mark it as always true or always false depending on what you want to test. For the real CLI run, you run a routine that performs the actual detection and stores it in the context.

For the check itself, it can look at `/proc/mounts` for the workspace fuse mount, it can check for `/.fuse-mounts`, it can check for `/databricks`, and I'm sure there are a couple other stable ways to determine this.

Reply:
Implemented this approach in https://github.com/databricks/cli/pull/1889.

Besides the environment variable, it also checks for the presence of a `/databricks` directory.